File: 백준/Silver/2606. 바이러스/바이러스.py
n = int(input())
m = int(input())
answer = 0

visited = [False] * (n+1)

# graph 채우기
graph = [[False] * (n+1) for _ in range(n+1)]
for _ in range(m):
  a, b = map(int, input().split())
  graph[a][b] = True
  graph[b][a] = True


def dfs(idx):
  global answer
  answer += 1
  visited[idx] = True
  for i in range(1, n+1):
    if not visited[i] and graph[idx][i]:
      dfs(i)

dfs(1)
print(answer - 1)

File: 백준/Silver/2606. 바이러스/바이러스.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function solution(N, input) {
  const graph = Array.from({ length: N + 1 }, () => []);
  const visited = Array(N + 1).fill(false);
  let answer = 0;

  for (const [a, b] of input) {
    graph[a].push(b);
    graph[b].push(a);
  }

  if (graph[1].length === 0) return 0;

  const q = [1];
  visited[1] = true;

  while (q.length > 0) {
    const node = q.shift();
    answer++;

    for (const newNode of graph[node]) {
      if (!visited[newNode]) {
        visited[newNode] = true;
        q.push(newNode);
      }
    }
  }
  return answer - 1;
}

let N = -1;
let M = -1;
let count = 0;
const input = [];

rl.on("line", (line) => {
  if (N === -1) {
    N = Number(line);
    return;
  }
  if (M === -1) {
    M = Number(line);
    if (M === 0) {
      console.log(0);
      rl.close();
    }
    return;
  }

  input.push(line.split(" ").map(Number));
  count++;

  if (count === M) {
    console.log(solution(N, input));
    rl.close();
  }
});
